doc/source/ray-core/handling-dependencies.rst

```diff
@@ -446,9 +446,63 @@ Runtime environment resources on each node (such as conda environments, pip pack
 
 When the cache size limit is exceeded, resources not currently used by any actor, task or job will be deleted.
 
+.. runtime-environments-job-conflict:
+
 Inheritance
 """""""""""
 
+**Job to Driver Inheritance**
+
+When both Ray Job and Driver specifies runtime environments, their runtime environments are merged if there's no conflict.
+Ray raises an exception if there's any conflicts in runtime environments. You can set an environment variable `RAY_OVERRIDE_JOB_RUNTIME_ENV=1` to override
+Job's runtime environment by Driver's runtime environment when there's a conflict. In this case, the merge behavior is equivalent to
+Driver to Tasks/Actors inheritance.
+
+* The ``runtime_env["env_vars"]`` field will be merged with the ``runtime_env["env_vars"]`` field of the parent.
+  This allows for environment variables set in the Job's runtime environment to be automatically propagated to the Driver, even if new environment variables are set in the Driver's runtime environment.
+  If there's any conflict in environment variables, Ray raises an exception.
+* If there's no conflict, every other field in the ``runtime_env`` will be merged.
+
+Example:
+
+.. testcode::
+
+  # Job's `runtime_env`
+  {"pip": ["requests", "chess"],
+  "env_vars": {"A": "a", "B": "b"}}
+
+  # Driver's specified `runtime_env`
+  {"env_vars": {"C": "c"}}
+
+  # Driver's actual `runtime_env` (merged with Job's)
```

Comment:
I assume by Driver's runtime env, you mean runtime_env specified in ray.init(), then the driver's actual runtime env can only be job's runtime env since ray.init runtime env only have effects on tasks and actors.

I'd avoid Driver's runtime_env, by saying runtime_env specified via ray.init()


Also there is another confusing thing: we normally say ray.init() runtime_env as job level runtime env but now there is another job level runtime env specified via job submission.

Reply:
I guess I will call it job-level runtime env specified ray job submit vs ray.init then

Reply:
I rewrote this part. Can you check?